src/components/learner-credit-management/cards/BaseCourseCard.jsx

```diff
@@ -0,0 +1,100 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import {
+  useMediaQuery,
+  breakpoints,
+  Card,
+  Stack,
+  Badge,
+} from '@edx/paragon';
+import { camelCaseObject } from '@edx/frontend-platform/utils';
+
+import useCourseCardMetadata from './data/useCourseCardMetadata';
+import CARD_TEXT from '../constants';
+
+const BaseCourseCard = ({
+  original,
+  footerActions: CardFooterActions,
+  enterpriseSlug,
+  ...rest
+}) => {
+  const isSmall = useMediaQuery({ maxWidth: breakpoints.small.maxWidth });
+  const isExtraSmall = useMediaQuery({ maxWidth: breakpoints.extraSmall.maxWidth });
+  const courseCardMetadata = useCourseCardMetadata({
+    course: camelCaseObject(original),
+    enterpriseSlug,
+  });
```

Comment:
Most of the business logic within `CourseCard` was abstracted out into a custom hook `useCourseCardMetadata` to keep the component itself focused on the UI and the hook focused on the metadata backing the card's UI. Improves the component's maintainability and readability.